sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationPropertySource.java

```diff
@@ -49,12 +92,227 @@ public Object getProperty(String name) {
+            List<Object> groupRollouts = convertToListOrEmptyList(parameters, GROUPS_CAPS);
+
+            switchKeyValues(parameters, USERS_CAPS, USERS, mapValuesByIndex(users));
+            switchKeyValues(parameters, GROUPS_CAPS, GROUPS, mapValuesByIndex(groupRollouts));
+            switchKeyValues(parameters, DEFAULT_ROLLOUT_PERCENTAGE_CAPS, DEFAULT_ROLLOUT_PERCENTAGE,
+                parameters.get(DEFAULT_ROLLOUT_PERCENTAGE_CAPS));
+
+            featureFilterEvaluationContext.setParameters(parameters);
+            featureEnabledFor.put(filter, featureFilterEvaluationContext);
+            feature.setEnabledFor(featureEnabledFor);
+
+        }
+        return feature;
+
+    }
+
+    /**
+     * Looks at each filter used in a Feature Flag to check what types it is using.
+     * 
+     * @param featureFlag FeatureFlagConfigurationSetting
+     * @param tracing The TracingInfo for this store.
+     */
+    protected void updateTelemetry(FeatureFlagConfigurationSetting featureFlag, TracingInfo tracing) {
+        for (FeatureFlagFilter filter : featureFlag.getClientFilters()) {
+            tracing.getFeatureFlagTracing().updateFeatureFilterTelemetry(filter.getName());
+        }
+    }
+
+    protected void processConfigurationSettings(List<ConfigurationSetting> settings, String keyFilter,
+        List<String> trimStrings, AppConfigurationKeyVaultClientFactory keyVaultClientFactory)
+        throws JsonProcessingException {
+        for (ConfigurationSetting setting : settings) {
+            if (trimStrings == null && StringUtils.hasText(keyFilter)) {
+                trimStrings = new ArrayList<>();
+                trimStrings.add(keyFilter.substring(0, keyFilter.length() - 1));
+            }
+            String key = trimKey(setting.getKey(), trimStrings);
+
+            if (setting instanceof SecretReferenceConfigurationSetting) {
+                String entry = getKeyVaultEntry(keyVaultClientFactory, (SecretReferenceConfigurationSetting) setting);
+
+                // Null in the case of failFast is false, will just skip entry.
+                if (entry != null) {
+                    properties.put(key, entry);
+                }
+            } else if (StringUtils.hasText(setting.getContentType())
+                && JsonConfigurationParser.isJsonContentType(setting.getContentType())) {
+                Map<String, Object> jsonSettings = JsonConfigurationParser.parseJsonSetting(setting);
+                for (Entry<String, Object> jsonSetting : jsonSettings.entrySet()) {
+                    key = trimKey(jsonSetting.getKey(), trimStrings);
+                    properties.put(key, jsonSetting.getValue());
+                }
+            } else {
+                properties.put(key, setting.getValue());
+            }
+        }
+    }
+
+    protected void processConfigurationSettingsSnapshot(List<ConfigurationSetting> settings, List<String> trimStrings,
```

Comment:
Maybe we could move the `processConfigurationSettingsSnapshot`  method to "AppConfigurationSnapshotPropertySource.java".
It's a method related to snapshot and it's only called in "AppConfigurationSnapshotPropertySource.java"

Reply:
This is no longer needed after the redesign.